cypress/integration/home-spec.js

```diff
@@ -0,0 +1,130 @@
+        cy.visit('localhost:3000')
+        cy.scrollTo('bottom')
+        cy.get('.home-button').click()
+        .window().its('scrollY').should('equal', 0)
+    })
+
+    it('If the movie posters data is unavailable (server error), the user should see an error message.', () => {
+        cy.intercept('GET', 'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+            statusCode: 500,
+            ok: false,
+            body: {},
+        })
+        cy.visit('localhost:3000')
+        cy.get('.error').contains('Apologies!')
+    })
+})
+
+describe('Rancid Tomatillos search flow', () => {
+
+    beforeEach(() => {
+        cy.fixture('movies.json').as('movies')
+        cy.intercept('GET',  'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+            statusCode: 200,
+            ok: true,
+            body: { movies },
+        })
+        cy.visit('localhost:3000')
+    })
+
+    it('When the user types into the search input, the movies are filtered by the value of the input.', () => {
+        cy.get('.text-input').type('u')
+        cy.get('.poster-grid').children().should('have.length', 2)
+        cy.get('.text-input').type('l')
+        cy.get('.poster-grid').children().should('have.length', 1)
+        cy.get('.text-input').clear()
+        cy.get('.poster-grid').children().should('have.length', 3)
+    })
+    
+    it('When the user deletes an input, the movies are filtered by the new value of the input.', () => {
+        cy.get('.text-input').type('ul')
+        cy.get('.poster-grid').children().should('have.length', 1)
+        cy.get('.text-input').type('{backspace}')
+        cy.get('.poster-grid').children().should('have.length', 2)
+        cy.get('.text-input').type('{backspace}')
+        cy.get('.poster-grid').children().should('have.length', 3)
+    })
+    
+    it('When typing into the search input, the URL should reflect the value of the search.', () => {
+        cy.get('.text-input').type('mul')
+            .url().should('include', '?search=mul')
+    })
+})
+
+describe('Rancid Tomatillos sort flow', () => {
+
+    beforeEach(() => {
+        cy.fixture('movies.json').as('movies')
+        cy.intercept('GET',  'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+            statusCode: 200,
+            ok: true,
+            body: { movies },
+        })
+        cy.visit('localhost:3000')
+    })
+    
+    it('When the user selects the filter for Rating: high to low, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('descendingRating')
+        cy.get('.poster-grid').find('img').first()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//uOw5JD8IlD546feZ6oxbIjvN66P.jpg')
+        cy.get('.poster-grid').find('img').last()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//c59eplVELdwrUfGBUAZVin3HfaL.jpg')
```

Comment:
I was really curious to see how you went about this, and here it is!  .first() and .last() with a .find(). Will definitely be referencing these test files later.  @annamkummer just casually printing out our testing files for mid-mod "notes" 😆

Reply:
Not intended to scare you with that comment.  Mid-mod and finals never go that in-depth!